### PR TITLE
switch from CircleCI -> local release process

### PR DESCRIPTION
**Pull request by JKillian**
**Warning: When this PR is merged `NPM_TOKEN` should be deleted from the CircleCI environment variables.**

- [x] Related to https://github.com/palantir/tslint/issues/2985
- [x] Documentation update

#### Overview of change:
Switches from a GH/CircleCI based publishing flow to a local one. This allows us to add contributors to TSLint without giving them publishing/releasing permissions. 

#### Is there anything you'd like reviewers to focus on?
Does the local publishing script seem reasonable? Does re-cloning the repo to a temporary directory seem reasonable? I tested the script myself, but obviously didn't publish an actual release.

Possible improvements to this PR:
* We could probably automate the creation of the git tag as well. For now, I decided to leave it as a manual step (since you have to paste the changelog into GH anyways).
* I only test the build locally with one version of TS. We could do what we do on CircleCI and make `publish:local` run the tests with multiple versions of TS.

### CODE REVIEW

**Comment by IllusionMH** [comment]
In theory it is possible to use [Circle CI API](https://circleci.com/docs/api/v1-reference/#recent-builds-project) call to confirm that build for supposed release commit hash is successful.

For multiple versions tests I tried to use Docker, but on Windows there is problems on linking step during yarn install (no problems with npm). However Mac may not have this problem.

**Comment by JKillian** [comment]
Both interesting ideas @IllusionMH! I'll leave those experiments for a future PR though, this one is primarily to unblock us so we can add collaborators. Publishing improvements will definitely be possible though 😁

File: README.md [hunk 1]
```diff
@@ -68,7 +68,8 @@ Creating a new release
...
 1. Bump the version number in `package.json` and `src/linter.ts`
 2. Add release notes in `CHANGELOG.md`
-3. `yarn verify` to build the latest sources from a clean state
 4. Commit with message `Prepare release <version>`
-5. Run `npm publish`
-6. Create a git tag for the new release and push it ([see existing tags here](https://github.com/palantir/tslint/tags))
+5. Push your branch to GitHub and make a PR
+6. Once your PR is merged, wait for the tests to pass on CircleCI for develop
+7. Create a "Release" on GitHub with the proper tag version and notes from the changelog
+8. Run 'yarn run publish:local 4.1.0' (replace with the appropriate tag name for the release)
```
Comment (adidahiya) [inline README.md]: I thought you need `--` to forward arguments to the script?

File: scripts/npmPublish.sh [hunk 2]
```diff
@@ -1,10 +1,33 @@
...
+# Publishes TSLint to NPM
+# This script must be run with yarn: "yarn run publish:local gitTagToPublish"
+# A user running this script must have Palantir NPM organization credentials
+
 set -e
 
-echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
-chmod 0600 .npmrc
+if [[ $# -ne 1 ]]; then
+    echo "usage: yarn run publish:local gitTagToPublish"
```
Comment (adidahiya) [inline scripts/npmPublish.sh]: can't we grab this from `echo 'console.log("./package.json").version' | node` or something similar?

File: README.md (hunk 1 above)
Comment (JKillian) [inline README.md]: I believe that's only needed if you're trying to pass dash options (like passing `-o` to your script or something like that)

File: scripts/npmPublish.sh (hunk 2 above)
Comment (JKillian) [inline scripts/npmPublish.sh]: hmm good point, I actually use `$npm_package_version` below

**Comment by adidahiya** [comment]
I've deleted `$NPM_TOKEN` from CircleCI.

**Pull request merged**